packages/wonder-blocks-dropdown/components/multi-select-menu.test.js

```diff
@@ -72,48 +72,41 @@ describe("MultiSelectMenu", () => {
-        expect(currentlySelected.length).toEqual(2);
-        expect(currentlySelected.includes("1")).toEqual(true);
-        expect(currentlySelected.includes("2")).toEqual(true);
-        expect(currentlySelected.includes("3")).toEqual(false);
-        // Now manually set the selectedValues like clients would
-        menu.setProps({selectedValues: ["1", "2"]});
-
-        // This menu should still be open afer a selection
-        expect(menu.state("open")).toEqual(true);
-
-        // Select all of the items
-        const selectAll = menu.find(ActionItem).at(0);
-        selectAll.simulate("mousedown");
-        selectAll.simulate("mouseup");
-        selectAll.simulate("click");
-        expect(allChanges.pop().length).toEqual(3);
-
-        // Select none of the items
-        const selectNone = menu.find(ActionItem).at(1);
-        selectNone.simulate("mousedown");
-        selectNone.simulate("mouseup");
-        selectNone.simulate("click");
-        expect(allChanges.pop().length).toEqual(0);
-
-        // Menu should still be open
-        expect(menu.state("open")).toEqual(true);
+        // expect(menu.state("open")).toEqual(true);
+        // expect(menu.prop("selectedValues")).toEqual(["2"]);
+        // // Grab the second item in the list
+        // const item = menu.find(SelectItem).at(0);
+        // expect(item.text()).toEqual("item 1");
+        // // Click the item 2, deselecting it
+        // item.simulate("mousedown");
+        // item.simulate("mouseup");
+        // item.simulate("click");
+        // // Expect menu's onChange callback to have been called
+        // expect(onClick).toHaveBeenCalledTimes(1);
+        // expect(allChanges.length).toEqual(1);
+        // const currentlySelected = allChanges.pop();
+        // // Expected selected items are now 1 and 2
+        // expect(currentlySelected.length).toEqual(2);
+        // expect(currentlySelected.includes("1")).toEqual(true);
+        // expect(currentlySelected.includes("2")).toEqual(true);
+        // expect(currentlySelected.includes("3")).toEqual(false);
+        // // Now manually set the selectedValues like clients would
+        // menu.setProps({selectedValues: ["1", "2"]});
+        // // This menu should still be open afer a selection
+        // expect(menu.state("open")).toEqual(true);
+        // // Select all of the items
+        // const selectAll = menu.find(ActionItem).at(0);
+        // selectAll.simulate("mousedown");
+        // selectAll.simulate("mouseup");
+        // selectAll.simulate("click");
+        // expect(allChanges.pop().length).toEqual(3);
+        // // Select none of the items
+        // const selectNone = menu.find(ActionItem).at(1);
+        // selectNone.simulate("mousedown");
+        // selectNone.simulate("mouseup");
+        // selectNone.simulate("click");
+        // expect(allChanges.pop().length).toEqual(0);
+        // // Menu should still be open
+        // expect(menu.state("open")).toEqual(true);
```

Comment:
This test case now gives me trouble because I cannot grab the portaled React elements to test and verify callbacks on. Dispatching events on the base `HTMLElement` does not invoke the callbacks on the corresponding components.

Reply:
Hmm... @somewhatabstract thoughts on how to get these to work?

Reply:
A few thoughts:
- code that's shared between tests should appear in a `beforeEach` not inside of the `describe`.  The reason being that the code inside `describe` is only run once and you'll want the setup code to run once per `it`.
- you're doing a full `mount` which is good.  You'll have to grab a ref to the dropdown being rendered in the portal.  See https://github.com/Khan/wonder-blocks/blob/cbbd25522b105b4a22ae0a8fbb1132fad9d168b5/packages/wonder-blocks-tooltip/components/tooltip.test.js for details.

Reply:
Almost never put code in the `describe`. If you want something to run once before all tests use `beforeAll`, or `afterAll` for something to run once after all tests. If you want something to run before or after each test, use `beforeEach` or `afterEach` instead.

Reply:
As @kevinbarabash said, you'll want to use a `ref` prop to grab the dropdown being rendered into the portal as in the example he linked.

Reply:
Since React portals preserve the React tree, I think now I can change the test case back and it'd work.